packages/browser/src/integrations/spotlight.ts

```diff
@@ -0,0 +1,100 @@
+import { getNativeImplementation } from '@sentry-internal/browser-utils';
+import { defineIntegration } from '@sentry/core';
+import type { Client, Envelope, Event, IntegrationFn } from '@sentry/types';
+import { logger, serializeEnvelope } from '@sentry/utils';
+import type { WINDOW } from '../helpers';
+
+import { DEBUG_BUILD } from '../debug-build';
+
+export type SpotlightConnectionOptions = {
+  /**
+   * Set this if the Spotlight Sidecar is not running on localhost:8969
+   * By default, the Url is set to http://localhost:8969/stream
+   */
+  sidecarUrl?: string;
+};
+
+export const INTEGRATION_NAME = 'SpotlightBrowser';
+
+const _spotlightIntegration = ((options: Partial<SpotlightConnectionOptions> = {}) => {
+  const sidecarUrl = options.sidecarUrl || 'http://localhost:8969/stream';
+
+  return {
+    name: INTEGRATION_NAME,
+    setupOnce: () => {
+      /* Empty function to ensure compatibility w/ JS SDK v7 >= 7.99.0 */
+    },
+    setup: () => {
+      DEBUG_BUILD && logger.log('Using Sidecar URL', sidecarUrl);
+    },
+    processEvent: event => {
+      // We don't want to send interaction transactions/root spans created from
+      // clicks within Spotlight to Sentry. Neither do we want them to be sent to
+      // spotlight.
+      if (isSpotlightInteraction(event)) {
+        return null;
+      }
+
+      if (event.type || !event.exception || !event.exception.values) {
+        return event;
+      }
+
+      return event;
+    },
+    afterAllSetup: (client: Client) => {
```

Comment:
m: we need to add the logic to force install all integrations if no DSN is set by users.While I haven't tested if this works, given we handle it a bit differently in the spotlight overlay where we inject the integration but `afterAllSetup` should be called late enough in the SDK lifecycle so that we can install the integrations within this hook.